src/clients/EvmClient.ts

```diff
@@ -52,6 +52,52 @@ export class EvmClient {
     });
   }
 
+  public isExecuted(commandId: string) {
+    return this.gateway.isCommandExecuted(commandId);
+  }
+
+  // warning: this function should be called after the command is executed, otherwise it will always return false
+  public isCallContractExecuted(
```

Comment:
No, in this application context, `Executed` means the relayer has called the `destination` contract. On the other hand, `Approved` means the relayer has called the `gateway` contract to obtain approval before it's able to call the destination contract. These terms are similar way to how they are used in Axelarscan.
![image](https://user-images.githubusercontent.com/78221556/229975281-674a1fcd-e132-46c7-826a-2a1744b13ef9.png)

Reply:
the function queries ```this.gateway
      .isContractCallApproved```, a bit confused

Reply:
Yes, this is bit confused. The gateway's `isContractCallApproved` function is actually used to check whether if the destination contract has already been executed, instead of checking gateway contract approval. I'm not sure how can we make this less confusing since we cannot change the gateway function name.

Fyi, there're 3 scenarios for the return data of the `isContractCallApproved` function:
- If it returns true, it means the relayer has already sent approval tx to the gateway contract.
- if it returns false, it can be two things: 1. the relayer hasn't sent approval tx to the gateway contract yet. 2. the relayer has already sent approval tx to the gateway contract + called the destination contract.